docs/preview/yamlgettingstarted-triggers.md

```diff
@@ -0,0 +1,96 @@
+# YAML getting started - YAML triggers (not yet available, for discussion only)
+
+## Simple CI trigger syntax
+
+On push, the YAML file from the branch that was updated, is used to determine whether
+to queue a CI build.
+
+For example, if master is updated, the trigger defined in master will be used. If a different
+branch is updated, the YAML file from that branch will be used to determine whether to queue a
+build.
+
+```yaml
+trigger:
+- master
+- releases/*
+```
+
+## CI is opt-out
+
+When not specified, the default trigger for build would be all branches.
+
+For build this behavior makes sense. Especially when we start supporting on-push defintion creation from any branch. And on-push resource authorization from any branch.
+
+For release, CI trigger would likely be opt-in? However with YAML, release will be tightly coupled with a repository. So opt-out CI behavior might also make sense for release? Or opt-out, but the default for RM only includes master?
+
+```yaml
+trigger: false | none # Turn off CI
+```
+
+## Full CI trigger syntax
+
+```yaml
+trigger:
+  batch: bool | number
+  branches:
+    include: []
+    exclude: []
+  paths:
+    include: []
+    exclude: []
+```
+
+## Schedules (future; needs cleanup)
+
+```yaml
+schedules:
```

Comment:
I think this is okay, but I would allow for a cron string too.  This is pretty common in the OSS world and has a lot of useful shortcuts.  They also allow for 'star syntax', which means that times will randomly get picked.  This causes workloads to be spread evenly over the day or hour.  This is very useful for triggers that run often.  Examples:
```
@daily - Runs every day at some point (same point every day)
@hourly - Runs every hour, at some point in the hour (same point every hour)
0 0 10 ? * 6L - Fires at 10:00 the last friday of every month.
```

Reply:
We have discussed CRON at the moment the job scheduler in VSTS does not support that syntax so we would have to do something custom.  We have talked about ideas and maybe we could support a CRON like syntax but not necessarily everything right away.